File: backend/app/schemas/category.py
from decimal import Decimal
from pydantic import Field
from .base import BaseSchema

class ProductRef(BaseSchema):
    id: int
    name: str
    price: Decimal

class CategoryCreate(BaseSchema):
    name: str

class CategoryUpdate(BaseSchema):
    name: str

class CategoryRef(BaseSchema):
    id: int
    name: str

class CategoryResponse(CategoryRef):
    active: bool

class CategoryWithProducts(CategoryResponse):
    products: list[ProductRef] = Field(default_factory=list)

File: frontend/src/types/category.ts
import type { Product } from "./product"

/**
 * Interface de categorías
 */
export interface Category {
  id: number
  name: string
  active: boolean
}

export interface CategoryCreate {
  name: string
}

export interface CategoryUpdate {
  name: string
}

export interface CategoryWithProducts {
  id: number
  name: string
  active: boolean
  products: Product[]
}